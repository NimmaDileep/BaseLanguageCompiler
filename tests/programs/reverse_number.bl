// For Loops

fun rev_num(num: Int){
    var rev = 0;

    for( ;num != 0; num = num/10){
        var rem = num % 10;
        rev = rev * 10 + rem;
    }
    return rev;
}

var num1 = 123456;
print(rev_num(num1));

var num2 = 0;
print(rev_num(num2));

var num3 = 325;
print(rev_num(num3));

var num4 = -5956;
print(rev_num(num4));