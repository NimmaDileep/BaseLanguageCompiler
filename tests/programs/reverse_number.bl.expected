6543210
0
523
-6595